src/tox/session/env_select.py

```diff
@@ -315,7 +315,8 @@ def _mark_active(self) -> None:
         if labels or factors:
             for env_info in self._defined_envs_.values():
                 env_info.is_active = False  # if any was selected reset
-            if labels:
+            # ignore labels when provisioning will occur
+            if labels and (self._provision is None or not self._provision[0]):
```

Comment:
While at it should we do the same for factors?

Reply:
for the factors piece, the code iterates through the environments, setting the attribute on the object(s) it gets back, so there's no risk of dereferencing an env name that doesn't exist. Above in `_build_env`, we're already NOT building the envs when provisioning is active, so the factors loop would be iterating through a mostly empty `self._defined_envs_`